server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -932,19 +934,21 @@ public IndexResult index(Index index) throws IOException {
                     final Translog.Location location;
                     if (indexResult.getResultType() == Result.Type.SUCCESS) {
                         location = translogManager.add(new Translog.Index(index, indexResult));
-                    } else if (indexResult.getSeqNo() != SequenceNumbers.UNASSIGNED_SEQ_NO) {
-                        // if we have document failure, record it as a no-op in the translog and Lucene with the generated seq_no
-                        final NoOp noOp = new NoOp(
-                            indexResult.getSeqNo(),
-                            index.primaryTerm(),
-                            index.origin(),
-                            index.startTime(),
-                            indexResult.getFailure().toString()
-                        );
-                        location = innerNoOp(noOp).getTranslogLocation();
-                    } else {
-                        location = null;
-                    }
+                    } else if (indexResult.getSeqNo() != SequenceNumbers.UNASSIGNED_SEQ_NO
```

Comment:
This also expects  indexResult.getFailure() to be not null apart from not instance check of the new exception.
Should we change it to following (as used in line 962) to be safe ?
```
!(indexResult.getFailure() != null && (indexResult.getFailure() instanceof IndexingRetryException))

```

Same for the other check down at line 962 as well

Reply:
This is to ensure for an indexing retry operation, we do not add a translog entry.